aspnetcore/mvc/advanced/custom-formatters/Sample/Formatters/VcardInputFormatter.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using CustomFormatterDemo.Models;
+
+namespace CustomFormatterDemo.Formatters
+{
+    #region classdef
+    public class VcardInputFormatter : TextInputFormatter
+    #endregion
+    {
+        #region ctor
+        public VcardInputFormatter()
+        {
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("text/vcard"));
+
+            SupportedEncodings.Add(Encoding.UTF8);
+            SupportedEncodings.Add(Encoding.Unicode);
+        }
+        #endregion
+
+        #region canreadtype
+        protected override bool CanReadType(Type type)
+        {
+            if (type == typeof(Contact))
+            {
+                return base.CanReadType(type);
```

Comment:
Need not call base here as the base always would return true. Can change all this to just `return type == typeof(Contact);`